llvm/include/llvm/Target/GlobalISel/Combine.td

```diff
@@ -1036,7 +1037,20 @@ def sdiv_by_const : GICombineRule<
    [{ return Helper.matchSDivByConst(*${root}); }]),
   (apply [{ Helper.applySDivByConst(*${root}); }])>;
 
-def intdiv_combines : GICombineGroup<[udiv_by_const, sdiv_by_const]>;
+def sdiv_by_pow2 : GICombineRule<
+  (defs root:$root),
+  (match (G_SDIV $dst, $x, $y, (MIFlags (not IsExact))):$root,
```

Comment:
Are there any tests showing that your combine does nothing on an exact div?

Reply:
I added two in `sdiv.i32.ll` and `sdiv.i64.ll`.